nova/app.py

```diff
@@ -24,30 +26,86 @@
             'margin': '10px 0'
         }
     ),
-    
+
     html.Div(id='output-numbers'),
     html.Div(id='output-image'),
 ])
 
 def extract_data(image):
     content_type, content_string = image.split(',')
+    pil_image = Image.open(io.BytesIO(base64.b64decode(content_string)))
+
+    crop_coordinates_list = [
+        (281,367,359,381),  # Crop 1 coordinates
+        (258,905,295,920),  # Crop 2 coordinates
+        (318,905,356,920),  # Crop 3 coordinates
+        (258,934,295,951),  # Crop 4 coordinates
+        (319,936,356,953),  # Crop 5 coordinates
+        (156,936,224,953),  # Crop 6 coordinates
+        (156,968,224,983),  # Crop 7 coordinates
+        (302,1104,357,1119) # Crop 8 coordinates
+    ]
+
+    extracted_data = {}
+
+    keys = [
+        "eye:",
+        "pupil_center_x:",
+        "pupil_center_y:",
+        "pachy_apex_x:",
+        "pachy_apex_y:",
+        "pachy_apex_thickness:",
+        "thinnest_local_thickness:",
+        "pupil_diameter:"
+    ]
 
-    # Do something with the image (content_string)
+    for idx, crop_coordinates in enumerate(crop_coordinates_list):
+        cropped_image = pil_image.crop(crop_coordinates)
+        
+        # Perform OCR on the cropped image region
+        extracted_text = pytesseract.image_to_string(cropped_image, config='--psm 6').strip()
+        
+        # Handle missing - sing
+        if idx == 1 or idx == 2:
+            if '+' not in extracted_text:
+                extracted_text = '-' + extracted_text
+
+        # Handle text recognition common errors
+        extracted_text = extracted_text.replace("p", "")
+        extracted_text = extracted_text.replace("u", "")
+        extracted_text = extracted_text.replace("m", "")
+        extracted_text = extracted_text.replace("o", "0")
+        extracted_text = extracted_text.replace("O", "0")
+        extracted_text = extracted_text.replace("+0", "+0.")
+        extracted_text = extracted_text.replace("+0..", "+0.")
+        extracted_text = extracted_text.replace("-0", "-0.")
+        extracted_text = extracted_text.replace("-0..", "-0.")
+        extracted_text = extracted_text.replace("+1", "+1.")
+        extracted_text = extracted_text.replace("+1..", "+1.")
+        extracted_text = extracted_text.replace("-1", "-1.")
+        extracted_text = extracted_text.replace("-1..", "-1.")
+        extracted_text = extracted_text.replace("+2", "+2.")
+        extracted_text = extracted_text.replace("+2..", "+2.")
+        extracted_text = extracted_text.replace("-2", "-2.")
+        extracted_text = extracted_text.replace("-2..", "-2.")
+
+        # Hande the occuring of additional . error
+        if len(extracted_text) > 5:
+            extracted_text = extracted_text[:-1]
```

Comment:
Can you move this code into a function and write unit tests for that function? 
The code here is dodgy asf, let's make sure it does what it is supposed to do.